doc/rest-api/Client-frontend.md

```diff
@@ -13,68 +13,53 @@ Please see the [Authentication](#authentication) section for more details.
 1. The relevant endpoint has to be configured on the server side.
 See the [configuration section](#configuration).
 1. A list of provided actions is documented with Swagger.
-See the beatiful [specification](http://mongooseim.readthedocs.io/en/latest/swagger/index.html?client=true).
+See the beautiful [specification](http://mongooseim.readthedocs.io/en/latest/swagger/index.html?client=true).
 
 ## Authentication
 
 The only possible authentication method for the time being is *Basic Authentication*.
-The *userid* part is user's *bare JID* and the password is the same as that used to
+The *userid* part is user's *bare JID* and the password are the same as the ones used to
 register the user's account.
 
 ### Bare JID
 
-To ilustrate what bare JIDs are, let's assume your MongooseIM server's hostname is
+To illustrate what bare JIDs are, let's assume your MongooseIM server's hostname is
 *wonderland.com* and the user is *alice*.
 In this case the bare JID for her is just: *alice@wonderland.com*.
 This value should be used as the *userid* in the Basic Authentication method for all the REST API calls.
 
 ## Configuration
 
-In order to enable the REST API, the following configuration should be added to the
-*listen* section in *mongooseim.cfg* file.
-
-```erlang
-  { 8089 , ejabberd_cowboy, [
-      {num_acceptors, 10},
-      {max_connections, 1024},
-      {compress, true},
-      {ssl, [{certfile, "priv/ssl/fake_cert.pem"}, {keyfile, "priv/ssl/fake_key.pem"}, {password, ""}]},
-      {modules, [
-          {"_", "/api/messages/[:with]", mongoose_client_api_messages, []},
-          {"_", "/api/rooms/:id/messages",    mongoose_client_api_rooms_messages, []},
-          {"_", "/api/rooms/:id/users/[:user]",    mongoose_client_api_rooms_users, []},
-          {"_", "/api/rooms/[:id]",    mongoose_client_api_rooms, []}
-      ]}
-  ]}
-```
-
-The most important part of the above example is the *modules* lists where the relevant
-REST API functionalities are enabled and exposed on the given paths.
-By default the REST API is exposed on port 8089 but this can be changed to whatever is more convenient.
-
-For more details about possible `ejabberd_cowboy` configuration parameters please
-see the relevant documentation in the [Listener modules](../advanced-configuration/Listener-modules.md#http-based-services-bosh-websocket-rest-ejabberd_cowboy).
+In order to enable the REST API, a listener has to be configured in the
+`mongooseim.toml` file.
 
+Functionalities have to be enabled and exposed for handlers on given paths, as 
+shown in the [REST API configuration example](../advanced-configuration/listen.md#example-3-client-api).
```

Comment:
`Handlers have to be configured as shown in the [REST API configuration example] to enable REST API.`